x/cork/keeper/keeper.go

```diff
@@ -0,0 +1,272 @@
+package keeper
+
+import (
+	"bytes"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	paramtypes "github.com/cosmos/cosmos-sdk/x/params/types"
+	mapset "github.com/deckarep/golang-set"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/peggyjv/sommelier/x/cork/types"
+	"github.com/tendermint/tendermint/libs/log"
+)
+
+// Keeper of the oracle store
+type Keeper struct {
+	storeKey      sdk.StoreKey
+	cdc           codec.BinaryCodec
+	paramSpace    paramtypes.Subspace
+	stakingKeeper types.StakingKeeper
+	gravityKeeper types.GravityKeeper
+}
+
+// NewKeeper creates a new distribution Keeper instance
+func NewKeeper(
+	cdc codec.BinaryCodec, key sdk.StoreKey, paramSpace paramtypes.Subspace,
+	stakingKeeper types.StakingKeeper, gravityKeeper types.GravityKeeper,
+) Keeper {
+	// set KeyTable if it has not already been set
+	if !paramSpace.HasKeyTable() {
+		paramSpace = paramSpace.WithKeyTable(types.ParamKeyTable())
+	}
+
+	return Keeper{
+		storeKey:      key,
+		cdc:           cdc,
+		paramSpace:    paramSpace,
+		stakingKeeper: stakingKeeper,
+		gravityKeeper: gravityKeeper,
+	}
+}
+
+// Logger returns a module-specific logger.
+func (k Keeper) Logger(ctx sdk.Context) log.Logger {
+	return ctx.Logger().With("module", "x/"+types.ModuleName)
+}
+
+///////////////////////
+// MsgSubmitCork //
+///////////////////////
+
+// SetCork sets the prevote for a given validator
+// CONTRACT: must provide the validator address here not the delegate address
+func (k Keeper) SetCork(ctx sdk.Context, val sdk.ValAddress, cork types.Cork) {
+	bz := k.cdc.MustMarshal(&cork)
+	ctx.KVStore(k.storeKey).Set(types.GetCorkForValidatorAddressKey(val, common.HexToAddress(cork.Address)), bz)
+}
+
+// GetCork gets the prevote for a given validator
+// CONTRACT: must provide the validator address here not the delegate address
+func (k Keeper) GetCork(ctx sdk.Context, val sdk.ValAddress, cel common.Address) (types.Cork, bool) {
```

Comment:
Should this be following the "return nil and error" go pattern?

Reply:
I don't believe any Cosmos keeper functions follow that convention, I am following the current convention. @jackzampolin any idea why that is?